lib/osf-components/addon/components/institutions-widget/component.ts

```diff
@@ -18,18 +18,20 @@ import template from './template';
 @layout(template, styles)
 export default class InstitutionWidget extends Component.extend({
     loadNodeAffiliatedInstitutions: task(function *(this: InstitutionWidget) {
-        if (!this.node) {
+        if (!this.node || !this.shouldShowModal) {
             return undefined;
         }
 
         try {
-            const affiliatedList: QueryHasManyResult<Institution> =
-                yield this.node.queryHasMany('affiliatedInstitutions');
-            this.set('affiliatedList', affiliatedList);
+            const affiliatedList: QueryHasManyResult<Institution> = yield this.node.sparseLoadAll(
+                'affiliatedInstitutions',
+                { institution: ['id'] },
+            );
+            this.set('affiliatedList', affiliatedList.mapBy('id'));
         } catch (e) {
             return false;
         }
-    }),
+    }).on('didReceiveAttrs').restartable(),
```

Comment:
Instead of running this on `didReceiveAttrs`, probably better to put it in the `showModal` action (and also add that action back in). To handle the case where the node changes, could add a `didReceiveAttrs` that sets `shouldShowModal` false and clears `affiliatedList`.